pkg/sink/sink_test.go

```diff
@@ -928,3 +897,127 @@ func TestHandleEventWithInterceptorsAndTriggerAuth(t *testing.T) {
+	tb, tt := getResources(t, "$(body.repository.links.clone[1].href)")
+
+	authSA := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      userWithPermissions,
+			Namespace: userWithPermissions,
+			UID:       types.UID(userWithPermissions),
+		},
+	}
+
+	el := &triggersv1.EventListener{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "el",
+			Namespace: namespace,
+		},
+		Spec: triggersv1.EventListenerSpec{
+			Triggers: []triggersv1.EventListenerTrigger{{
+				ServiceAccount: &corev1.ObjectReference{
+					Namespace: userWithPermissions,
+					Name:      userWithPermissions,
+				},
+				Bindings: []*triggersv1.EventListenerBinding{{Name: "tb", Kind: "TriggerBinding"}},
+				Template: triggersv1.EventListenerTemplate{Name: "tt"},
+				Interceptors: []*triggersv1.EventInterceptor{{
+					Bitbucket: &triggersv1.BitBucketInterceptor{
+						SecretRef: &triggersv1.SecretRef{
+							SecretKey:  "secretKey",
+							SecretName: "secret",
+							Namespace:  namespace,
+						},
+						EventTypes: []string{"repo:refs_changed"},
+					},
+				}},
+			}},
+		},
+	}
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "secret",
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"secretKey": []byte("secret"),
+		},
+	}
+
+	resources := test.Resources{
+		TriggerBindings:  []*triggersv1.TriggerBinding{tb},
+		TriggerTemplates: []*triggersv1.TriggerTemplate{tt},
+		EventListeners:   []*triggersv1.EventListener{el},
+		Secrets:          []*corev1.Secret{secret},
+		ServiceAccounts:  []*corev1.ServiceAccount{authSA},
+	}
+	sink, dynamicClient := getSinkAssets(t, resources, el.Name, fakeAuth{})
+	ts := httptest.NewServer(http.HandlerFunc(sink.HandleEvent))
+	defer ts.Close()
+
+	triggerAuthWG.Add(1)
+
+	dynamicClient.PrependReactor("*", "*", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+		defer triggerAuthWG.Done()
+		return false, nil, nil
+	})
+
+	req, err := http.NewRequest("POST", ts.URL, bytes.NewReader(eventBody))
+	if err != nil {
+		t.Fatalf("Error creating Post request: %s", err)
+	}
+	req.Header.Add("X-Event-Key", "repo:refs_changed")
+	// This was generated by using SHA1 and hmac from go stdlib on secret and payload.
+	// https://play.golang.org/p/ovlZGxCP07E for a sample.
+	req.Header.Add("X-Hub-Signature", "sha1=efaa666dce82d7acd087779314c2f80eb3447102")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatalf("Error sending Post request: %v", err)
+	}
+
+	if resp.StatusCode != http.StatusCreated {
+		t.Fatalf("Response code doesn't match: expected status code %d vs. actual %d, entire statutes %v",
+			http.StatusCreated,
+			resp.StatusCode,
+			resp.Status)
+	}
+}
+
+func getResources(t *testing.T, value string) (*v1alpha1.TriggerBinding, *v1alpha1.TriggerTemplate) {
```

Comment:
nit: can we use a more descriptive name than value for the param?
Also, we should have a `t.Helper()` call in this function to indicate that this is a helper function